tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/rules/aux/CherryPick.scala

```diff
@@ -573,29 +574,54 @@ class CherryPick(rewriter: SymbStateRewriter) {
   }
 
   /**
-   * This is a new implementation of picking a function from a set. Since we started to encode functions
-   * as relations, the implementation became trivial
+   * Picks a function from a set.
    *
    * @param funType a cell type to assign to the picked cell.
-   * @param oracle  a variable that stores which element (by index) should be picked, can be unrestricted
-   * @param funs    a sequence of cells that store functions
-   * @param state   a symbolic state
+   * @param oracle  a variable that stores which element (by index) should be picked, can be unrestricted.
+   * @param funs    a sequence of cells that store functions.
+   * @param state   a symbolic state.
    * @return a new symbolic state with the expression holding a fresh cell that stores the picked element.
    */
   def pickFun(funType: FunT, state: SymbState, oracle: Oracle, funs: Seq[ArenaCell], elseAssert: TlaEx): SymbState = {
     rewriter.solverContext.log("; CHERRY-PICK %s FROM [%s] {".format(funType, funs.map(_.toString).mkString(", ")))
     var nextState = state
-    // pick the relation
-    val relationT = FinSetT(TupleT(Seq(funType.argType, funType.resultType)))
-    nextState = pickSet(relationT, nextState, oracle, funs map state.arena.getCdm, elseAssert)
-    val pickedRelation = nextState.asCell
-    // create a fresh cell to hold the function
-    nextState = nextState.setArena(nextState.arena.appendCell(funType))
-    val funCell = nextState.arena.topCell
-    val newArena = nextState.arena.setCdm(funCell, pickedRelation)
-    rewriter.solverContext.log(s"; } CHERRY-PICK $funCell:$funType")
-    // That's it! Compare to pickFunPreWarp.
-    nextState.setArena(newArena).setRex(funCell.toNameEx)
+    rewriter.solverContext.config.smtEncoding match {
+      case `arraysEncoding` =>
+        // Here we have a simplified version of pickBasic, using SMT equality instead of lazy equality.
```

Comment:
```suggestion
        // Here we have a simplified version of pickFun, using SMT equality instead of lazy equality.
```

Reply:
I actually mean `pickBasic` here. In `oopsla19` we use `pickSet` to pick a set of pairs, in `arrays` the goal is simply to pick a function cell from a list.